python/paddle/fluid/framework.py

```diff
@@ -1496,25 +1496,32 @@ def _optimized_guard(self, param_and_grads):
             >>> with program._optimized_guard([p,g]):
             >>>     p = p - 0.001 * g
         """
+        tmp_role = self._current_role
+        tmp_var = self._op_role_var
+
         OpRole = core.op_proto_and_checker_maker.OpRole
         self._current_role = OpRole.Optimize
         self._op_role_var = [
             var.name if isinstance(var, Variable) else var
             for var in param_and_grads
         ]
         yield
-        self._op_role_var = []
-        self._current_role = OpRole.Forward
+        self._op_role_var = tmp_var
+        self._current_role = tmp_role
 
     @contextlib.contextmanager
-    def _lr_schedule_guard(self):
+    def _lr_schedule_guard(self, is_with_opt=False):
```

Comment:
lr schedule ops can always be marked as `optimize | lrsched` I think, the learning rate should only be used by optimize ops.

Reply:
That is right, but note that **optimization is used to update the parameter, learning rate schedule is used to update the learning rate, and now every device only has one learning rate**.